### DISCUSSION TITLE
Question: event delegation vs attaching events right to DOM nodes

### DISCUSSION THREAD

**Discussion by boombang** [solidjs/solid#835] [Q&A] [answered]
Hi, I have a question regarding event listeners in Solid.

According to https://www.solidjs.com/tutorial/bindings_events we can either use event delegation via `onClick` or attach event listener directly to node via `on:click`. AFAIK `react` uses the first approach whereas `preact` -  the second one.

What are suitable cases for these approaches?
As I can see event delegation is intended to be used mostly. But when to use `on:click` then?

Big thanks in advance!

**Comment by atk** [comment, marked as answer]
Short answer: because performance!

Longer answer: binding and unbinding an event listener takes time. Processing different event listerens takes time. Having a single event listener that delegates the events to the nodes using it is a lot faster. The two main use cases are if a) you need the original event for whatever reason or b) you want a custom event that is not covered by on__.
